src/components/toolbox/calendar/monthView.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import 'moment/min/locales';
+import { validations, generateDayPlaceholder } from './calendarUtils';
+import styles from './calendar.css';
+
+class MonthView extends Component {
+  constructor(props) {
+    super();
+
+    this.options = {
+      dateFormat: props.dateFormat,
+      locale: props.locale,
+      amount: 'month',
+    };
+
+    this.nextMonth = this.nextMonth.bind(this);
+    this.previousMonth = this.previousMonth.bind(this);
+    this.showYearView = this.showYearView.bind(this);
+    this.handleSelectDate = this.handleSelectDate.bind(this);
+  }
+
+  previousMonth() {
+    if (validations.canGoToPrevious(this.props.showingDate, this.props.minDate, this.options)) {
+      const showingDate = moment(this.props.showingDate, this.props.dateFormat).subtract(1, 'month');
+      this.props.setShowingDate(showingDate);
+    }
+  }
+
+  nextMonth() {
+    if (validations.canGoToNext(this.props.showingDate, this.props.maxDate, this.options)) {
+      const showingDate = moment(this.props.showingDate, this.props.dateFormat).add(1, 'month');
+      this.props.setShowingDate(showingDate);
+    }
+  }
+
+  showYearView() {
+    this.props.setCurrentView('year');
+  }
+
+  handleSelectDate({ target }) {
+    this.props.onDateSelected(target.value);
+  }
+
+  render() {
+    const { locale, dateFormat, isShown } = this.props;
+    moment.locale(locale);
+    const selectedDate = moment(this.props.selectedDate, dateFormat);
+    const showingDate = moment(this.props.showingDate, dateFormat).startOf('month');
+    const day = moment(showingDate);
+
+    return (
+      <div className={`${!isShown ? styles.hidden : ''} monthView`}>
+        <header className={styles.calendarHeader}>
+          <span className={styles.navigationButton} onClick={this.previousMonth} />
+          <span
+            onClick={this.showYearView}
+            className={`${styles.viewName} ${styles.clickable}`}>
+            {showingDate.format('MMMM YYYY')}
+          </span>
+          <span className={styles.navigationButton} onClick={this.nextMonth} />
+        </header>
+        <div className={styles.contentWrapper}>
+          <div className={styles.monthHeader}>
+            {moment.weekdaysShort(true).map((weekday, key) =>
+              <div className={styles.weekday} key={key}>{weekday}</div>)}
+          </div>
+          <div className={styles.itemsContent}>
+            { generateDayPlaceholder(day.weekday(), moment(day).subtract(day.weekday(), 'days'), `${styles.item} ${styles.dayItem}`) }
+            { [...Array(day.daysInMonth())].map((_, d) => {
```

Comment:
Is it possible to put this chunk of code to separate function? Too much logic inside render return imo.